src/NumericalAlgorithms/Spectral/Spectral.cpp

```diff
@@ -372,6 +371,23 @@ Matrix interpolation_matrix(const size_t num_points,
   return interp_matrix;
 }
 
+template <size_t Dim>
+std::array<Matrix, Dim> interpolation_matrices(const Mesh<Dim>& mesh,
```

Comment:
How is this different from `RegularGrid` in `NumericalAlgorithms/Interpolation/RegularGridInterpolant.hpp`?